fleet/client.go

```diff
@@ -1,17 +1,61 @@
 package fleet
 
+import (
+	"io"
+	"net/http"
+	"net/url"
+)
+
 const (
 	basePath = "/fleet/v1"
 )
 
 type Client struct {
-	URL string
+	baseUrl       *url.URL
+	err           error
+	requestSender requestSender
 }
 
 // NewClient creates a basic client that is configured to be used
 // with the given fleet HTTP API URL.
-func NewClient(url string) *Client {
+func NewClient(rawUrl string) *Client {
+	baseUrl, err := url.ParseRequestURI(rawUrl)
+
+	if nil != err {
+		baseUrl = &url.URL{}
+	}
+
 	return &Client{
-		URL: url,
+		baseUrl:       baseUrl,
+		err:           err,
+		requestSender: &httpRequestSender{http.Client{}},
+	}
+}
+
+func NewValidClient(rawUrl string) (*Client, error) {
```

Comment:
Yeah I was thinking about it, but this will break backwards compatibility unfortunately. That's why I introduced new function.